### PR TITLE
Fix issue 174

### LINKED ISSUE #174: Provide batch operations for the Mesh CUDS API
```python

class ABCMesh(self):

    def add_points(iterable):
        """ Add a set of points from the provided iterable.

        """

    def update_points(iterable):
        """ Update a set of points from the provided iterable.

        """

    def remove_points(uids):
        """ Remove the points with the provided uids.

        """

    def add_edges(iterable):
        """ Add a set of edges from the provided iterable.

        """

    def update_edge(iterable):
        """ Update a set of edges from the provided iterable.

        """

    def remove_edges(uids):
        """ Remove the edges with the provided uids.

        """

    def add_faces(iterable):
        """ Add a set of faces from the provided iterable.

        """

    def update_faces(iterable):
        """ Update a set of faces from the provided iterable.

        """

    def remove_faces(uids):
        """ Remove the faces with the provided uids.

        """

    def add_cells(iterable):
        """ Add a set of cells from the provided iterable.

        """

    def update_cells(iterable):
        """ Update a set of cells from the provided iterable.

        """

    def remove_cells(uids):
        """ Remove the cells with the provided uids.

        """
```

**notes**:
- [ ] This will replace the `add_xxxx`, `update_xxxx` and `remove_xxxx` methods
- [ ] The Test templates need to be updated
- [ ] The Concrete Classes need to be updated
- [ ] Documentation needs to be updated

### PR DESCRIPTION
**Pull request by roigcarlo**
This PR implements batch ooperations for mesh and fixes #174 

- [x] This will replace the `add_xxxx`, `update_xxxx` and `remove_xxxx` methods
- [x] The Test templates need to be updated
- [x] Add test batch operations with multiple items 
- [x] The Concrete Classes need to be updated
- [x] Documentation needs to be updated

### CODE REVIEW

File: doc/source/uml/mesh.txt [hunk 1]
```diff
@@ -3,26 +3,26 @@ abstract class ABCMesh {
...
-  {abstract} update_edge(edge: Edge)
+  {abstract} update_edges(edge: iterable of Edge)
   {abstract} iter_edge(uids: iterable of uuid.UUID {optional}): iterator of Edge {unordered}
   {abstract} has_edges(): boolean
   ..
-  {abstract} add_face(face: Face): uuid.UUID
+  {abstract} add_faces(faces: iterable of Face): uuid.UUID
   {abstract} get_edge(uid: uuid.UUID): Face
-  {abstract} update_face(face: Face)
+  {abstract} update_faces(face: iterable of Face)
```
Comment (SGGgarcia) [inline doc/source/uml/mesh.txt]: face -> faces
In general, we need plural in all the methods except add_points and add_edges, which are correct in this PR.
Btw, in the case of particles I named the parameter of the new methods as `iterable`, we have to decide if its ok with iterable or if its better to make the names plural as you did, but I guess we should follow the same naming convention in all the places :+1:

File: simphony/cuds/abstractmesh.py [hunk 2]
```diff
@@ -30,35 +30,35 @@ def get_cell(self, uid):
...
         pass
 
     @abstractmethod
-    def update_edge(self, edge):
+    def update_edges(self, edge):
         pass
 
     @abstractmethod
-    def update_face(self, face):
+    def update_faces(self, face):
```
Comment (SGGgarcia) [inline simphony/cuds/abstractmesh.py]: Same comment as before with the naming convention

File: doc/source/uml/mesh.txt (hunk 1 above)
Comment (roigcarlo) [inline doc/source/uml/mesh.txt]: Maybe we can just call them ```iterator of Foo``` as done with the iter_xxxx methods. This should be consistent with the rest of the code.

Comment (itziakos) [inline doc/source/uml/mesh.txt]: I am in favor of `iterable`.

**Comment by itziakos** [comment]
@roigcarlo PR has conflicts with the current master

**Comment by itziakos** [comment]
:+1:

**Pull request merged**